lms/djangoapps/lti_provider/views.py

```diff
@@ -172,7 +173,12 @@ def get_optional_parameters(dictionary):
         original dictionary, or an empty dictionary if no optional parameters
         were present.
     """
-    return {key: dictionary[key] for key in OPTIONAL_PARAMETERS if key in dictionary}
+    custom_params = configuration_helpers.get_value("LTI_CUSTOM_PARAMETERS", settings.LTI_CUSTOM_PARAMETERS)
```

Comment:
Hey @xitij2000 I have an idea what if instead of doing these changes here we change the function definition to ```def get_optional_parameters(dictionary, additional=None): ```? The benefit of this would be that we can isolate this function and write test cases for it.

These changes can be pushed to the lti_launch function.

Reply:
I think that will just move the tests to a different place. Here it's more of a integrated test at the lti_launch level so individual functions don't need testing, making it easier to refactor internally. 

I will separate this function because that makes sense.